components/eventing-controller/cmd/eventing-controller/main.go

```diff
@@ -24,28 +25,27 @@ import (
 )
 
 func main() {
-	setupLogger := ctrl.Log.WithName("setup")
-
 	opts := options.New()
 	if err := opts.Parse(); err != nil {
-		setupLogger.Error(err, "parse options failed")
-		os.Exit(1)
+		log.Fatalf("Failed to parse options, error: %v", err)
 	}
 
 	ctrLogger, err := logger.New(opts.LogFormat, opts.LogLevel)
 	if err != nil {
-		setupLogger.Error(err, "initialize logger failed")
-		os.Exit(1)
+		log.Fatalf("Failed to initialize logger, error: %v", err)
 	}
 	defer func() {
 		if err := ctrLogger.WithContext().Sync(); err != nil {
-			setupLogger.Error(err, "flush logger failed")
+			log.Printf("Failed to flush logger, error: %v", err)
```

Comment:
Should it be `log.Errorf` ?

Reply:
Ideally yes, but the go-log doesn't have an equivalent to error, it has fatal which will also exit the program, the panic does the same. So the nearest one which will log the problem and continue the execution would be `Print`

Reply:
Can we then move this defer func definition after line 47, when we have an instance for `setupLogger`?